httpclient/api_client.go

```diff
@@ -129,9 +129,16 @@ func NewApiClient(cfg ClientConfig) *ApiClient {
 		rateLimit = rate.Inf
 		cfg.RetryTimeout = 0
 	}
+
+	burst := int(rateLimit)
+	// Ensure that burst is never 0
+	if burst < 1 {
+		burst = 1
+	}
+
 	return &ApiClient{
 		config:      cfg,
-		rateLimiter: rate.NewLimiter(rateLimit, 1),
+		rateLimiter: rate.NewLimiter(rateLimit, burst),
```

Comment:
FWIW, the `orDefault` call above is duplicated for `cfg.RateLimitPerSecond`.

It means the rate limit in this field is always > 0, and always an integer.

You can use `cfg.RateLimitPerSecond` directly.